src/test/java/io/confluent/kafkarest/integration/SimpleConsumerAvroTest.java

```diff
@@ -0,0 +1,182 @@
+  private final List<ProducerRecord<Object, Object>> recordsWithKeys = Arrays.asList(
+      new ProducerRecord<Object, Object>(
+          topicName, "key",
+          new GenericRecordBuilder(valueSchema).set("field", 72).build()),
+      new ProducerRecord<Object, Object>(
+          topicName, "key",
+          new GenericRecordBuilder(valueSchema).set("field", 73).build()),
+      new ProducerRecord<Object, Object>(
+          topicName, "key",
+          new GenericRecordBuilder(valueSchema).set("field", 74).build()),
+      new ProducerRecord<Object, Object>(
+          topicName, "key",
+          new GenericRecordBuilder(valueSchema).set("field", 75).build())
+  );
+
+  private static final GenericType<List<AvroConsumerRecord>> avroConsumerRecordType
+      = new GenericType<List<AvroConsumerRecord>>() {
+  };
+  private static final Converter converter = new Converter() {
+    @Override
+    public Object convert(Object obj) {
+      return AvroConverter.toJson(obj).json;
+    }
+  };
+
+
+  @Before
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    final int numPartitions = 1;
+    final int replicationFactor = 1;
+    TestUtils.createTopic(zkClient, topicName, numPartitions, replicationFactor,
+                          JavaConversions.asScalaIterable(this.servers).toSeq(), new Properties());
+  }
+
+  @Test
+  public void testConsumeOnlyValuesByOffset() {
+    produceAvroMessages(recordsOnlyValues);
+
+    simpleConsumeMessages(
+        topicName,
+        0,
+        null, // No "count" parameter in the query
+        recordsOnlyValues.subList(0, 1), // We expect only the first record in the response
+        Versions.KAFKA_V1_JSON_AVRO,
+        Versions.KAFKA_V1_JSON_AVRO,
+        avroConsumerRecordType,
+        converter
+    );
+  }
+
+  @Test
+  public void testConsumeWithKeysByOffset() {
+    produceAvroMessages(recordsWithKeys);
+
+    simpleConsumeMessages(
+        topicName,
+        0,
+        null, // No "count" parameter in the query
+        recordsWithKeys.subList(0, 1),
+        Versions.KAFKA_V1_JSON_AVRO,
+        Versions.KAFKA_V1_JSON_AVRO,
+        avroConsumerRecordType,
+        converter
+    );
+  }
+
+  @Test
+  public void testConsumeOnlyValuesByOffsetAndCount() {
+    produceAvroMessages(recordsOnlyValues);
+
+    simpleConsumeMessages(
+        topicName,
+        0,
+        recordsOnlyValues.size(),
+        recordsOnlyValues,
+        Versions.KAFKA_V1_JSON_AVRO,
+        Versions.KAFKA_V1_JSON_AVRO,
+        avroConsumerRecordType,
+        converter
+    );
+  }
+
+  @Test
+  public void testConsumeWithKeysByOffsetAndCount() {
+    produceAvroMessages(recordsWithKeys);
+
+    simpleConsumeMessages(
+        topicName,
+        0,
+        recordsWithKeys.size(),
+        recordsWithKeys,
+        Versions.KAFKA_V1_JSON_AVRO,
+        Versions.KAFKA_V1_JSON_AVRO,
+        avroConsumerRecordType,
+        converter
+    );
+  }
```

Comment:
Maybe add a test that checks non-zero offsets? I think that's the only parameter I don't see tested here.